lib/core_extensions.rb

```diff
@@ -39,8 +39,14 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
+        record.association(klass.to_sym).association_scope.each do |what|
```

Comment:
I would consider this fix for a stable branch of 1.9. as this allows the user to get a nasty foreign key SQL error.
therefore, IMHO we should have a fix for both rails 3 and 4.